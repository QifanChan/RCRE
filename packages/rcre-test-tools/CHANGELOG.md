# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.21.0](https://github.com/andycall/RCRE/compare/v0.20.14...v0.21.0) (2019-05-13)

**Note:** Version bump only for package rcre-test-tools
