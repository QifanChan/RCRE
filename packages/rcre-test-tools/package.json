{
    "name": "rcre-test-tools",
    "version": "0.21.11",
    "description": "rcre test utils",
    "main": "./dist/index.js",
    "typings": "./dist/index.d.ts",
    "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1"
    },
    "author": "andycall",
    "license": "MIT",
    "dependencies": {
        "@types/enzyme": "^2.8.12",
        "@types/pretty-format": "^20.0.0",
        "chalk": "^2.4.1",
        "json-format": "^1.0.1",
        "lodash": "^4.17.4",
        "pretty-format": "^23.6.0",
        "rcre": "^0.21.11"
    },
    "gitHead": "3cccf68d4c841aebe60ca2663fd523952591c859"
}
