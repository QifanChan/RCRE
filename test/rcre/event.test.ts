import {
    addEventListener,
    removeEventListener,
    ListenerFnItem,
    removeAllEventListener,
    removeAllEventListenerByEventName
} from 'rcre';
import {RCRETestUtil} from 'rcre-test-tools';
import moxios from 'moxios';
import axios from 'axios';

describe('Event', () => {
    beforeEach(() => {
        removeAllEventListener();
        moxios.install(axios);
    });

    afterEach(() => {
        moxios.uninstall(axios);
    });

    it('add logger', () => {
        addEventListener('RCRE_SET_DATA', (action, state) => {
            expect(action.type).toBe('RCRE_SET_DATA');
            expect(state.$rcre.container.demo.username).toBe('helloworld');
        });

        let config: any = {
            body: [{
                type: 'container',
                model: 'demo',
                children: [{
                    type: 'input',
                    name: 'username'
                }]
            }]
        };

        let test = new RCRETestUtil(config);
        test.setContainer('demo');
        let username = test.getComponentByName('username');

        test.setData(username, 'helloworld');
    });

    it('async middleware', async () => {
        addEventListener('RCRE_ASYNC_LOAD_DATA_SUCCESS', (action, state) => {
            expect(action.type).toBe('RCRE_ASYNC_LOAD_DATA_SUCCESS');
            expect(state.$rcre.container.demo).toMatchSnapshot();
        });

        let config: any = {
            body: [{
                type: 'container',
                model: 'demo',
                dataProvider: [{
                    mode: 'ajax',
                    namespace: 'test',
                    config: {
                        url: '/test',
                        method: 'GET',
                        data: {
                            username: '#ES{$data.username}'
                        },
                        requiredParams: ['username']
                    }
                }],
                children: [{
                    type: 'input',
                    name: 'username'
                }]
            }]
        };

        await new Promise((resolve) => {

            let test = new RCRETestUtil(config);
            test.setContainer('demo');
            moxios.wait(() => {
                let request = moxios.requests.mostRecent();

                setTimeout(() => {
                    request.respondWith({
                        status: 200,
                        response: {
                            errno: 0,
                            data: 'ok'
                        }
                    }).then(() => {
                        resolve();
                    });
                }, 100);
            });

            let username = test.getComponentByName('username');
            test.setData(username, 'helloworld');
        });
    });

    it('removeEventListener', () => {
        let setDataOne: ListenerFnItem = (action, state) => {
            expect(action.type).toBe('RCRE_SET_DATA');
            expect(state.$rcre.container.demo.username).toBe('helloworld');
        };

        let setDataTwo: ListenerFnItem = (action, state) => {
            throw new Error('error');
        };

        addEventListener('RCRE_SET_DATA', setDataOne);
        addEventListener('RCRE_SET_DATA', setDataTwo);
        removeEventListener('RCRE_SET_DATA', setDataTwo);

        let config: any = {
            body: [{
                type: 'container',
                model: 'demo',
                children: [{
                    type: 'input',
                    name: 'username'
                }]
            }]
        };

        let test = new RCRETestUtil(config);
        test.setContainer('demo');
        let username = test.getComponentByName('username');

        test.setData(username, 'helloworld');
    });

    it('removeAllEventListenerByEventName', () => {
        addEventListener('RCRE_SET_DATA', (action, state) => {
            throw new Error('error');
        });
        addEventListener('RCRE_SET_DATA', (action, state) => {
            throw new Error('error');
        });
        addEventListener('RCRE_SET_DATA', (action, state) => {
            throw new Error('error');
        });
        addEventListener('RCRE_SET_DATA', (action, state) => {
            throw new Error('error');
        });
        addEventListener('RCRE_SET_DATA', (action, state) => {
            throw new Error('error');
        });
        removeAllEventListenerByEventName('RCRE_SET_DATA');

        let config: any = {
            body: [{
                type: 'container',
                model: 'demo',
                children: [{
                    type: 'input',
                    name: 'username'
                }]
            }]
        };

        let test = new RCRETestUtil(config);
        test.setContainer('demo');
        let username = test.getComponentByName('username');

        test.setData(username, 'helloworld');
    });
});