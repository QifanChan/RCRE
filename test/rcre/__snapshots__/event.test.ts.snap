// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event async middleware 1`] = `
Object {
  "$error": null,
  "$loading": false,
  "test": Object {
    "data": "ok",
    "errno": 0,
  },
  "username": "helloworld",
}
`;
